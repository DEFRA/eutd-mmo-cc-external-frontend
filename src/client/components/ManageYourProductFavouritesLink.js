import React from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

const ManageYourProductFavouritesLink = ({styles}) => {
  const { t } = useTranslation();
  let text = t('ccProductFavouriteDetailSummary3ParagraphText');
  const manageYourProductFavouritesText = text.charAt(0).toUpperCase() + text.slice(1);

  return (
    <Link
      id="manageYourProductFavourites"
      to="/manage-favourites"
      style={styles}
    >
      {manageYourProductFavouritesText}
    </Link>
  );
};

ManageYourProductFavouritesLink.propTypes = {
  progressUri: PropTypes.string.isRequired,
  documentNumber: PropTypes.string.isRequired,
};

export default withRouter(ManageYourProductFavouritesLink);