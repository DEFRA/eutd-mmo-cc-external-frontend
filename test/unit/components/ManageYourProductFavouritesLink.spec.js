import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ManageYourProductFavouritesLink from '../../../src/client/components/ManageYourProductFavouritesLink';

describe('Manage your product favourites', () => {
  const { container } = render(
    <MemoryRouter>
      <ManageYourProductFavouritesLink/>
    </MemoryRouter>
  );

  it('should load and render the component', () => {
    expect(screen.getByText('Manage your product favourites')).toBeDefined();
    expect(container).toBeDefined();
  });

  it('should take a snapshot of the component', () => {
    const { container } = render(
      <MemoryRouter>
        <ManageYourProductFavouritesLink/>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
