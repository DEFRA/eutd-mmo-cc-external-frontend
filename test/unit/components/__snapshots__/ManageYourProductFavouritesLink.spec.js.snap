// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage your product favourites should take a snapshot of the component 1`] = `
<div>
  <a
    href="/manage-favourites"
    id="manageYourProductFavourites"
  >
    Manage your product favourites
  </a>
</div>
`;
